### PR TITLE
ensure all bytes to base 64 conversions are URL safe. Use UTF8 encodi…

### PR DESCRIPTION
**Pull request by lohstroh**
…ng of the body in the sample code. Allow for other methods to create the auth header.

### CODE REVIEW

**Review by bionoren — APPROVED** [review]

File: Modo2Auth.cs [hunk 1]
```diff
@@ -29,28 +35,22 @@ private String createSignature(String payload)
...
-        {
-            if (body == null)
-                body = "";
+        public String createModoToken(String uri, byte[] body) {
 
-            // Get a SHA of the request body 
-            byte[] hashBytes = _shaGenerator.ComputeHash(Encoding.ASCII.GetBytes(body));
-            String bodySha = BitConverter.ToString(hashBytes).Replace("-","").ToLower();
+            byte[] hashBytes = _shaGenerator.ComputeHash(body);;
+            String bodySha = BitConverter.ToString(hashBytes).Replace("-","").ToLower();    // also remove the - chars that c# adds in the conversion
```
Comment (adammodo) [inline Modo2Auth.cs:51]: **Kudos:** Thanks for including this explanation of baffling C# shenanigans.

File: Modo2Auth.cs [hunk 2]
```diff
@@ -29,28 +35,22 @@ private String createSignature(String payload)
...
-        public String createModoToken(String uri, String body="")
-        {
-            if (body == null)
-                body = "";
+        public String createModoToken(String uri, byte[] body) {
 
-            // Get a SHA of the request body 
-            byte[] hashBytes = _shaGenerator.ComputeHash(Encoding.ASCII.GetBytes(body));
-            String bodySha = BitConverter.ToString(hashBytes).Replace("-","").ToLower();
+            byte[] hashBytes = _shaGenerator.ComputeHash(body);;
```
Comment (adammodo) [inline Modo2Auth.cs]: **Nitpick:** Double semicolon looks messy.

**Review by adammodo — APPROVED** [review]
Only the tiniest of nitpicks here.

**Review by adammodo — APPROVED** [review]
This is a bit brute-force, and might get a little cranky under heavy load. But that's better than failing unpredictably. If client needs to build high traffic application, they should implement a resource pool based on their threading model.

**Pull request merged**